Automattic/wp-calypso
# Guided Tours: Find ongoing tour before selecting another

Fixes #6785 

# To test

1. Apply test commit, build. Note that this commit also disables `findOngoingTour`.
2. Launch Calypso at `/`. `MockTourOne` should pop up. **Quickly** press _Next_. That tour's second step should appear. Finish the tour. Moments later, `MockTourTwo` should start. This is a scenario that works without this PR's fix.
3. Repeat step 2 but *wait* (8 seconds, technically) before pressing _Next_ on the first tour. **Bug:** After pressing _Next_, instead of seeing the next step appearing, you should see the second tour suddenly starting. This is because the _context_ (`when`) for the first tour became invalid as time passed, but we don't want tours to unexpectedly change before completion.
4. Remove the short-circuit (`return null;`) in `state/ui/guided-tours/selectors#findOngoingTour`.
5. Repeat steps 2 and 3. This time, the first tour should never be abruptly swapped out for another.
